packages/core-ethereum/src/chain.ts

```diff
@@ -0,0 +1,21 @@
+import { ChainWrapper, createChainWrapper } from './ethereum'
+import { debug } from '@hoprnet/hopr-utils'
+
+const log = debug('hopr:core-ethereum:chain')
+
+export default class ChainWrapperSingleton {
```

Comment:
Why don't make this a function in `ethereum.ts`?

Reply:
I need this to be exported as a default package for `sinon.createStubInstance` to work properly when other packages (e.g. `HoprEthereum`) call it internally, the same way we can stub `Libp2p` static methods.

I'm also a big fan of `1 file = 1 class` Java-like approach, we have too many files doing too many things that no longer represent a unit of work but rather a concept we need to contextualize people with.

Reply:
Solving this one for now, please let me know otherwise @robertkiel

Reply:
Possibly for a follow-up PR: Missing some documentation about the classes intended use and reasoning as already mentioned in the previous comments.